lld/ELF/LinkerScript.cpp

```diff
@@ -555,6 +583,32 @@ LinkerScript::computeInputSections(const InputSectionDescription *cmd,
   // Matched sections after the last SORT* are sorted by (--sort-alignment,
   // input order).
   sortByPositionThenCommandLine(sizeAfterPrevSort, ret.size());
+
+  // The flag --enable-non-contiguous-regions may cause sections to match an
+  // InputSectionDescription in more than one OutputSection. Matches after the
+  // first were collected in the spills set, so replace these with potential
+  // spill sections.
+  if (!spills.empty()) {
+    for (InputSectionBase *&sec : ret) {
+      if (!spills.contains(sec))
+        continue;
+
+      PotentialSpillSection *pss = make<PotentialSpillSection>(
+          sec, const_cast<InputSectionDescription *>(cmd));
+
```

Comment:
The code style is pretty dense. Variables are not followed by a blank line.

`auto [it, inserted] = ..` is preferred for newer code